packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -307,25 +336,35 @@ export class GCFBootstrapper {
 
           // TODO: find out the best way to get this type, and whether we can
           // keep using a custom event name.
-          await this.probot.receive({
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            name: name as any,
-            id,
-            payload: body,
-          });
+          const span = tracer.startSpan('execute probot');
+          try {
+            await this.probot.receive({
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              name: name as any,
+              id,
+              payload: body,
+            });
+          } finally {
+            span.end();
+          }
         } else if (triggerType === TriggerType.SCHEDULER) {
           // TODO: currently we assume that scheduled events walk all repos
           // managed by the client libraries team, it would be good to get more
           // clever and instead pull up a list of repos we're installed on by
           // installation ID:
           await this.handleScheduled(id, request, name, signature, wrapOptions);
         } else if (triggerType === TriggerType.GITHUB) {
-          await this.enqueueTask({
-            id,
-            name,
-            signature,
-            body: JSON.stringify(request.body),
-          });
+          const span = tracer.startSpan('enqueue task');
+          try {
+            await this.enqueueTask({
+              id,
+              name,
+              signature,
+              body: JSON.stringify(request.body),
+            });
+          } finally {
+            span.end();
```

Comment:
is `span.end` synchronous? wondering if there's a way to await all spans flushing?

Reply:
This just marks the span as ended (writes the end timestamp). Completed spans are published when the publisher is force flushed.

Reply:
I'm not a fan of these try/finally blocks - most other OpenCensus/OpenTelemetry implementations use callable closures to ensure the spans are closed - but I see how it's an issue with the `async` functions.